src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php

```diff
@@ -51,6 +56,12 @@ public static function getPublicAccessSimulationCode(
             $target = '$this->' . $valueHolder->getName();
         }
 
+        if ($interfaceName !== null) {
+            return '$targetObject = ' . $target . ';' . "\n\n"
+                . self::getUndefinedPropertyNotice($operationType, $nameParameter, $interfaceName)
+                . self::getOperation($operationType, $nameParameter, $valueParameter);
+        }
+
         return '$realInstanceReflection = new \\ReflectionClass(get_parent_class($this));' . "\n\n"
```

Comment:
without this change, this code is hit: `get_parent_class($this)` returns `false`, which is cast to the empty string by `ReflectionClass`, and a fatal error occurs: `Class  does not exist`.